packages/connectivity_plus/connectivity_plus/android/build.gradle

```diff
@@ -18,22 +18,18 @@ rootProject.allprojects {
         google()
         mavenCentral()
     }
-    tasks.withType(JavaCompile).configureEach {
-    javaCompiler = javaToolchains.compilerFor {
-      languageVersion = JavaLanguageVersion.of(8)
-    }
-  }
-}
-
-project.getTasks().withType(JavaCompile){
-    options.compilerArgs.addAll(args)
 }
```

Comment:
I am not sure if we should remove this part! any reasons you have removed lines 28 and 29?

Reply:
I thought it was part of the bug, but looks like it works with those lines

Reply:
nevermind, I don't really understand why but those lines cause other bugs. Yesterday I think it was working because  the cache was not refreshed

Reply:
Here is the stacktrace I have with them, maybe it'll help understand:
```
Launching lib/main.dart on Android SDK built for x86 in debug mode...
/Users/terence/.pub-cache/hosted/pub.dartlang.org/background_fetch-1.0.1/android/src/main/java/com/transistorsoft/flutter/backgroundfetch/BackgroundFetchPlugin.java:9: warning: [deprecation] Registrar in PluginRegistry has been deprecated
import io.flutter.plugin.common.PluginRegistry.Registrar;
                                              ^

error: warnings found and -Werror specified
/Users/terence/.pub-cache/hosted/pub.dartlang.org/background_fetch-1.0.1/android/src/main/java/com/transistorsoft/flutter/backgroundfetch/BackgroundFetchPlugin.java:16: warning: [deprecation] Registrar in PluginRegistry has been deprecated
    public static void registerWith(Registrar registrar) {
                                    ^
/Users/terence/.pub-cache/hosted/pub.dartlang.org/background_fetch-1.0.1/android/src/main/java/com/transistorsoft/flutter/backgroundfetch/BackgroundFetchPlugin.java:25: warning: [deprecation] PluginRegistrantCallback in PluginRegistry has been deprecated
    public static void setPluginRegistrant(PluginRegistry.PluginRegistrantCallback callback) {
                                                         ^

/Users/terence/.pub-cache/hosted/pub.dartlang.org/background_fetch-1.0.1/android/src/main/java/com/transistorsoft/flutter/backgroundfetch/HeadlessTask.java:41: warning: [deprecation] PluginRegistrantCallback in PluginRegistry has been deprecated
    private static PluginRegistry.PluginRegistrantCallback sPluginRegistrantCallback;
                                 ^
/Users/terence/.pub-cache/hosted/pub.dartlang.org/background_fetch-1.0.1/android/src/main/java/com/transistorsoft/flutter/backgroundfetch/HeadlessTask.java:55: warning: [deprecation] PluginRegistrantCallback in PluginRegistry has been deprecated
    static void setPluginRegistrant(PluginRegistry.PluginRegistrantCallback callback) {
                                                  ^
1 error
5 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':background_fetch:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
BUILD FAILED in 1m 29s
Exception: Gradle task assembleDebug failed with exit code 1
Exited (sigterm)

```